Provide Dockerfile for initial MNTN-related testing.

sfalvo-aerospike opened the pull request:
This Dockerfile builds a Debian Trixie base image with the latest Rust installed.  It then executes a Bash script named "startup-sequence", whose job it is to run the `cargo test` command to invoke tests.  Note that some environment variables are required when running the resulting Docker image (see Rust client docs).

As more property-based testing facilities are provided, this Dockerfile and startup-sequence may be amended to take additional configuration via environment variables.

sfalvo-aerospike committed:
Provide Dockerfile for initial MNTN-related testing.

This Dockerfile builds a Debian Trixie base image with the latest Rust
installed.  It then executes a Bash script named "startup-sequence",
whose job it is to run the `cargo test` command to invoke tests.  Note
that some environment variables are required when running the resulting
Docker image (see Rust client docs).

As more property-based testing facilities are provided, this Dockerfile
and startup-sequence may be amended to take additional configuration via
environment variables.

juliannguyen4 reviewed (approved):
LGTM, this is fine since we need to get the rests running ASAP. Just something to note in the future:

- We should add a README explaining what the Dockerfile is used for
- Also you can `chmod u+x` the startup script and commit those permissions in the repo
- There's an official image for Rust: https://hub.docker.com/_/rust
- To test the Dockerfile, you can use it to run the tests in CI/CD

sfalvo-aerospike committed:
Fix branch to be tls and not v2.

v2 is an older branch than tls; tls is where all mainline development is
being staged for MNTN.

khaf commented on tools/docker/Dockerfile:
```diff
@@ -0,0 +1,13 @@
+
+COPY startup-sequence /app/startup-sequence
+RUN chmod +x /app/startup-sequence
+RUN echo 'source $HOME/.cargo/env' >> $HOME/.bashrc
```
This is already sourced in the script, why add it here?

khaf commented on tools/docker/startup-sequence:
```diff
@@ -0,0 +1,11 @@
+
+git clone https://github.com/aerospike/aerospike-client-rust.git
+cd aerospike-client-rust
+git checkout tls
```
I believe the name of the branch should be an ENV variable.

khaf commented on tools/docker/startup-sequence:
```diff
@@ -0,0 +1,11 @@
+cd aerospike-client-rust
+git checkout tls
+cargo clean
+cargo fix --lib -p aerospike-core
```
Should be removed. Fixes should be applied to the source, otherwise tests will run on different source code.

khaf commented on tools/docker/startup-sequence:
```diff
@@ -0,0 +1,11 @@
+git checkout tls
+cargo clean
+cargo fix --lib -p aerospike-core
+cargo test
```
Where do the ENV variables come from? IIRC they need to be explicitly passed here, no?

juliannguyen4 commented:
Also noticed that Rust client code seems to be building when running the container. Shouldn't this already be built in the image?

sfalvo-aerospike commented:
> Also noticed that Rust client code seems to be building when running the container. Shouldn't this already be built in the image?

This is what I was talking about yesterday regarding the `cargo test --no-run` flag.  However, I wasn't able to get it working in the Dockerfile.  So, rather than commit a broken Dockerfile, I did the "worse-is-better" thing and just let Cargo build the tests before running them.  We can always iterate and improve upon this later.

sfalvo-aerospike committed:
Add environment variable log block

sfalvo-aerospike committed:
Provide default ENV settings that can be overridden via -e option

khaf committed:
Batch proptest scaffolding

sfalvo-aerospike committed:
WIP

sfalvo-aerospike committed:
Randomize AEROSPIKE_NAMESPACE to avoid problems in long tests.

Prior to this patch, AEROSPIKE_NAMESPACE is left unspecified, and so
defaults to just "test".  This can cause problems with multiple
concurrently running client containers running against a common cluster.

After applying this patch, each running docker image has its own unique
setting, thus isolating each container to a unique namespace.

sfalvo-aerospike committed:
Complete addition of read_ops to the batch read tests.

This was a MONSTER to get right, but I eventually got it!

sfalvo-aerospike committed:
Support batch write tests, I think.

I'm not confident I implemented this correctly at all.  Khosrow, can you
please review my work here?  Thanks!

khaf committed:
Fix Batch encoding issue

khaf committed:
Allow Set name tp be set in prop tests via AEROSPIKE_PROP_SET_NAME

sfalvo-aerospike committed:
Merge branch 'sam_proptests' into dockerfile-for-client-tests

This brings in the recent changes to allow set name to be parameterized
instead of the namespace.

sfalvo-aerospike committed:
Change environment to randomize set name instead of namespace.

juliannguyen4 committed:
Set to false by default

juliannguyen4 committed:
Remove all env var overrides
